backend/application/advertisement.ts

```diff
@@ -15,12 +14,72 @@ export class AdvertisementApplicationService {
           imageUrl,
           author,
           startFrom,
-          endAt
-        }
+          endAt,
+        },
       })
       return advertisement
     } catch (error) {
-      throw new Error(`AdvertisementApplicationService: create advertisement error: ${error}`)
+      throw new Error(
+        `AdvertisementApplicationService: create advertisement error: ${error}`,
+      )
+    }
+  }
+  async getAdvertisement(id: string) {
+    try {
+      const advertisement = await prisma.advertisement.findUnique({
+        where: { id },
+      })
+      return advertisement
+    } catch (error) {
+      throw new Error(
+        `AdvertisementApplicationService: get advertisement error: ${error}`,
+      )
+    }
+  }
+  async getAdvertisements() {
+    try {
+      const advertisement = await prisma.advertisement.findMany({
+        where: {
+          startFrom: {
+            lte: new Date(), // 現在の日付以前の開始日時
+          },
+        },
+      })
+      return advertisement
+    } catch (error) {
+      throw new Error(
+        `AdvertisementApplicationService: get advertisements error: ${error}`,
+      )
+    }
+  }
+
+  async updateAdvertisement(params: {
+    id: string
+    name: string
+    url: string
+    imageUrl: string
+    author: string
+    startFrom: Date
+    endAt: Date
+  }) {
+    try {
+      const { id, name, url, imageUrl, author, startFrom, endAt } = params
+      const updateAdvertisement = await prisma.advertisement.update({
+        where: { id },
+        data: {
+          name,
+          url,
+          imageUrl,
+          author,
+          startFrom,
+          endAt,
+        },
+      })
+      return updateAdvertisement
```

Comment:
```suggestion
      const updatedAdvertisement = await prisma.advertisement.update({
        where: { id },
        data: {
          name,
          url,
          imageUrl,
          author,
          startFrom,
          endAt,
        },
      })
      return updatedAdvertisement
```

過去分詞の更新されたというのを示す
edをつけましょう！細いですが確認お願いします！